app/src/main/java/com/arielwang/workoutlogger/features/addexerciseflow/track/ui/screen/TrackViewModel.kt

```diff
@@ -163,6 +195,38 @@ class TrackViewModel @Inject constructor(
         _uiState.value = viewState
     }
 
+    // the value of TextField cannot be negative
+    private fun noNegativeNumber(text: String): String {
```

Comment:
This function is doing more than it's name indecates. It calculates the value first, then check if it's less than 0. But the name only says it checks if the text is less than 0. It would be better to move the `text.toInt() - 1` outside this function

Reply:
Thank you for your opinion.